drivers/pinmux/stm32/pinmux_stm32g0.h

```diff
@@ -11,21 +12,15 @@
 /**
  * @file Header for STM32G0 pin multiplexing helper
  */
-#define STM32G0_PINMUX_FUNC_PA2_USART2_TX \
-	(STM32_PINMUX_ALT_FUNC_1 | STM32_PUSHPULL_NOPULL)
-#define STM32G0_PINMUX_FUNC_PA3_USART2_RX \
-	(STM32_PINMUX_ALT_FUNC_1 | STM32_PUPDR_NO_PULL)
 
-#define STM32G0_PINMUX_FUNC_PA9_USART1_TX \
-	(STM32_PINMUX_ALT_FUNC_1 | STM32_PUSHPULL_NOPULL)
-#define STM32G0_PINMUX_FUNC_PB6_USART1_TX \
-	(STM32_PINMUX_ALT_FUNC_0 | STM32_PUSHPULL_NOPULL)
-#define STM32G0_PINMUX_FUNC_PB7_USART1_RX \
-	(STM32_PINMUX_ALT_FUNC_0 | STM32_PUPDR_NO_PULL)
-#define STM32G0_PINMUX_FUNC_PB6_USART1_TX_RX \
-	(STM32_PINMUX_ALT_FUNC_0 | STM32_OPENDRAIN_PULLUP)
+#if defined(CONFIG_SOC_STM32G071XX)
+/*This section is for STM32G071XX or STM32G081XX*/
+#include "pinmux_stm32g071x.h"
+#elif defined(CONFIG_SOC_STM32G031XX)
+/* This section is for STM32G031XX or STM32G041XX */
+#include "pinmux_stm32g041x.h"
+#else
+#error "CONFIG_GPIO_STM32 enabled, but unknown GO type"
+#endif  /* SOC_STM32G071XX */
```

Comment:
I'm a little bit worried that you're duplicating the whole definitions (while I'm sure we're far from 100% difference)
My main concern is that anyone wanting to introduce a new G0 variant not referenced will miss G0 basic pinmux definitions.
This is in contradiction with the approach in use that ease the porting of new variants by providing already available definitions when matching the SoC.

Reply:
@erwango would it be sufficient if i also added support for the 2 missing series parts x30 and x40? Then the entire series would be covered. 
Right now kconfig options are missing for the additional parts anyways.

Reply:
I split things out to common definitions, and per part variants, and made the formatting a little nicer.